core/src/main/java/com/linecorp/armeria/internal/DefaultTimeoutController.java

```diff
@@ -76,100 +85,184 @@ public void setTimeoutTask(TimeoutTask timeoutTask) {
      *
      * <p>Note that the {@link TimeoutTask} should be set via the {@link #setTimeoutTask(TimeoutTask)} or
      * the {@link #DefaultTimeoutController(TimeoutTask, EventLoop)} before calling this method.
+     *
+     * @return {@code true} if the timeout is scheduled.
+     *         {@code false} if the timeout has been scheduled, the timeout has been triggered already
+     *         or the {@link TimeoutTask} could not be scheduled now.
      */
     @Override
-    public void initTimeout(long timeoutMillis) {
-        if (timeoutFuture != null || timeoutTask == null || !timeoutTask.canSchedule()) {
-            // No need to schedule a response timeout if:
-            // - the timeout has been scheduled already,
-            // - the timeout task is not set yet or
-            // - the timeout task cannot schedule yet or
-            return;
+    public boolean scheduleTimeout(long timeoutMillis) {
+        checkArgument(timeoutMillis > 0,
+                      "timeoutMillis: " + timeoutMillis + " (expected: > 0)");
+        ensureInitialized();
+        if (state != State.DISABLED || !timeoutTask.canSchedule()) {
+            return false;
         }
 
+        cancelTimeout();
         this.timeoutMillis = timeoutMillis;
-        firstStartTimeNanos = lastStartTimeNanos = System.nanoTime();
-        if (timeoutMillis > 0) {
-            timeoutFuture = eventLoop.schedule(timeoutTask, timeoutMillis, TimeUnit.MILLISECONDS);
-        }
+        state = State.SCHEDULED;
+        timeoutFuture = eventLoop.schedule(this::invokeTimeoutTask, timeoutMillis, TimeUnit.MILLISECONDS);
+        return true;
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * <p>Note that the {@link TimeoutTask} should be set via the {@link #setTimeoutTask(TimeoutTask)} or
+     * the {@link #DefaultTimeoutController(TimeoutTask, EventLoop)} before calling this method.
+     *
+     * @return {@code true} if the current timeout is extended by the specified {@code adjustmentMillis}.
+     *         {@code false} if no timeout was scheduled previously, the timeout has been triggered already
+     *         or the {@link TimeoutTask} could not be scheduled now.
```

Comment:
How about: `{@link TimeoutTask#canScheduled()} returned {@code false}.` to make it more direct and clear?